src/sentry/search/utils.py

```diff
@@ -64,6 +65,14 @@ def parse_status_value(value: Union[str, int]) -> int:
     raise ValueError("Invalid status value")
 
 
+def parse_substatus_value(value: Union[str, int]) -> int:
+    if value in SUBSTATUS_UPDATE_CHOICES:
+        return int(SUBSTATUS_UPDATE_CHOICES[value])
+    if value in SUBSTATUS_UPDATE_CHOICES.values():
+        return int(value)
```

Comment:
```suggestion
def parse_substatus_value(key_or_value: Union[str, int]) -> int:
    if key_or_value in SUBSTATUS_UPDATE_CHOICES.keys():
        return int(SUBSTATUS_UPDATE_CHOICES[key_or_value])
    if key_or_value in SUBSTATUS_UPDATE_CHOICES.values():
        return int(key_or_value)
```

I know using `.keys()` is redundant, but IMHO it makes it a touch more obvious what these two logical branches are about.